### ISSUE TITLE
Url.Decode(string s, bool interpretPlusAsSpace)

### ISSUE THREAD

**Issue by lazybnz** [tmenier/Flurl#437]
Bug in method Url.Decode(string s, bool interpretPlusAsSpace)
'+' character decoded after UnescapeDataString, so any plus character in value (encoded as %2B) is lost

**Comment by tmenier** [comment]
Can you provide a 2 or 3 line code sample to demonstrate? Just want to make sure I understand correctly.

**Comment by lazybnz** [comment]
```
// query param "q=param_with_+"
var urlStr = "http://google.com/search?q=param_with_%2B";
var url = new Url(urlStr);
var paramValue = url.QueryParams["q"];
Console.WriteLine($"q={paramValue}");
//output "q=param_with_ "
```
replacement of '+' should be done before UnescapeDataString

**Comment by tmenier** [comment]
Yep, this looks legit. Thanks for reporting.

**Comment by tmenier** [comment]
This is fixed and released in Flurl 2.8.2. Thanks again for reporting.
